torch/csrc/jit/ir/ir.h

```diff
@@ -1512,5 +1527,77 @@ struct OperatorSet {
   std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>> ops;
 };
 
+template <typename T>
+struct OperatorMap {
+  // Type aliasing
+  using OpMapType = typename std::pair<std::shared_ptr<Operator>, T>;
+  using ValueType = std::vector<OpMapType>;
+  using MapType = std::unordered_map<Symbol, ValueType>;
+
+  OperatorMap() = default;
+  explicit OperatorMap(
+      std::initializer_list<std::pair<std::shared_ptr<Operator>, T>> init) {
+    insert(init);
+  }
+
+  void insert(const std::shared_ptr<Operator>& op, T val) {
+    map[Symbol::fromQualString(op->schema().name())].emplace_back(
```

Comment:
should we enforce no duplicates ? the existing `OperatorSet` doesn't do it but that's because it's a set and not a map... Maybe we should erase and then insert here ?